scripts/populate_expressions_list.py

```diff
@@ -0,0 +1,206 @@
+        text += format_variant(variant, f_name)
+    return text
+
+def format_variant(function_dict, f_name):
+
+    # Organize arguments
+    arg_syntax_list = []
+    arg_description_list = []
+    optional_args = False
+    variable_args = ''
+    if 'arguments' in function_dict:
+        for arg in function_dict['arguments']:
+            if "optional" in arg and arg["optional"]:
+                if "default" in arg:
+                    default = f'={arg["default"]}'
+                else:
+                    default = ''
+                arg_text = f'[{arg["arg"]}{default}]'
+                optional_args = True
+            else:
+                arg_text = arg['arg']
+            if 'syntaxOnly' in arg:
+                arg_syntax_list.append(arg_text)
+                variable_args = ', ...'
+            elif 'descOnly' in arg:
+                arg_description_list.append(f"**{arg['arg']}** - {sphynxify_html(arg['description'],9)}")
+            else:
+                arg_syntax_list.append(arg_text)
+                arg_description_list.append(f"**{arg['arg']}** - {sphynxify_html(arg['description'],9)}")
+
+    # Prepare syntax and arguments strings
+    syntax = f'   * - Syntax\n     - '
+    if len(arg_syntax_list) > 0:
+        syntax += f'{f_name}({", ".join(arg_syntax_list)}{variable_args})\n'
+        descriptions = '\n       * '.join(arg_description_list)
+        arguments = (f"   * - Arguments\n"
+                     f"     - * {descriptions}\n"
+                     f"\n")
+    elif f_name.startswith("$"):
+        syntax += f"{f_name}\n"
+        arguments = ''
+    else:
+        syntax += f"{f_name}()\n"
+        arguments = ''
+
+    if optional_args:
+        syntax += '\n       [] marks optional arguments\n'
+
+    if 'type' in function_dict and function_dict['type'] == 'expression':
+        syntax = ''
+
+    # Prepare examples
+    if 'examples' in function_dict:
+        ex_list = [f"{ex['expression']} → {ex['returns']}" for ex in function_dict['examples']]
+        examples = "   * - Examples\n     - * "+ "\n\n       * ".join(ex_list)
+    else:
+        examples = ''
+
+    if 'variant' in function_dict:
+        v_description = (f"**{function_dict['variant']}**\n\n"
+                         f"{function_dict['variant_description']}\n\n")
+    else:
+        v_description = ''
+
+    if 'notes' in function_dict:
+        notes = f"\n\n.. note:: {function_dict['notes']}"
+    else:
+        notes = ''
+
+    text = (f"{v_description}"
+            f".. list-table::\n"
+            f"   :widths: 15 85\n"
+            f"   :stub-columns: 1\n"
+            f"\n"
+            f"{syntax}"
+            f"{arguments}"
+            f"{examples}"
+            f"{notes}"
+            f"\n\n\n")
+    return text
+
+# Get a list of all files on that folder
+
+(_, _, filenames) = next(walk(help_folder))
+
+# Store groups and functions in memory
+groups = {}
```

Comment:
```suggestion
groups = {}
groups['Form'] = {'description': 'Form group', 'func_list': []}
groups['Processing'] = {'description': 'Processing group', 'func_list': []}
groups['Layout'] = {'description': 'Layout group', 'func_list': []}
```
These are all referenced by functions, but I can't find them defined. I guess it will be useful to also have them defined, so that we can use the functions that reference them (*Form*: current_parent_value,current_value; *Processing*: parameter; *Layout*: item_variables).

Reply:
@havatv  That's because, at the same time, I have been fixing the JSON file on my local machine, all changes are now here:

https://github.com/qgis/QGIS/pull/37837

To fully test this script for master, you need to fetch my PR too.

Reply:
I will ignore this suggestion as this will be properly addressed by adding these groups help file in qgis/QGIS#37837